Add teleport request + response & UI modal

SahilP20 opened the pull request:
**Implemented Teleport Buttons/UI for sending and receiving:** (Backend not implemented. User won't actually move to a friend's location upon accepted teleport request.)

**Key Features:**
- Teleport Button implemented to allow a user to teleport to a person on their friend's list:
  <img width="371" height="149" alt="image" src="https://github.com/user-attachments/assets/74181cf3-c92c-4750-9142-54ba6bbb6249" />

- A message appears for the person sending a teleport request to notify them that the request has been sent successfully:
  <img width="371" height="149" alt="image" src="https://github.com/user-attachments/assets/aa756812-16ec-4369-9067-8883f7c25bab" />

- The friend receiving a teleport request has a UI pop-up appear in the middle of their screen where they can accept/deny the teleport request.

  <img width="371" height="149" alt="image" src="https://github.com/user-attachments/assets/726ee226-bb0c-4a72-9388-5e1935f2184d" /> 

- If the friend accepts the teleport request the following message appears for them:
  <img width="371" height="149" alt="image" src="https://github.com/user-attachments/assets/3d5b512d-a3e1-419b-8990-10ef05695ac6" />

SahilP20 committed:
Add teleport request + response & UI modal

dathomir01 commented:
Looks good, just a minor text issue. The teleport request text implies that the person accepting the request is the one that's going to teleport.

SahilP20 committed:
Update Profile.tsx

Fixed "Teleporting to" to "Teleporting" as player teleports to user that accepts the request

SahilP20 commented:
> Looks good, just a minor text issue. The teleport request text implies that the person accepting the request is the one that's going to teleport.

Fixed it.
